[BUG]: Response status code always shows as 200 in TypeScript types

### What happened?

For the `GET /repos/{owner}/{repo}/issues/{issue_number}/comments` endpoint (and possibly also others), [the Octokit docs](https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28#list-issue-comments-for-a-repository--status-codes) list 200, 404, and 422 as possible status codes. However, in practice, `response.status` is always typed as 200, so any attempt to compare it to another value generates a TypeScript error:

<img width="1996" height="442" alt="Checking response.status gives the error This comparison appears to be unintentional because the types '200' and '404' have no overlap.ts(2367)" src="https://github.com/user-attachments/assets/acebc5fe-c32b-4e78-bcc0-79693d874227" />

On a related note, the status codes in the docs appear to be incomplete. The [GitHub API docs](https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28&versionId=free-pro-team%40latest&restPage=using-pagination-in-the-rest-api#exceeding-the-rate-limit) separately state that you may get a 403 or 429:

> If you exceed your primary rate limit, you will receive a 403 or 429 response, and the x-ratelimit-remaining header will be 0. You should not retry your request until after the time specified by the x-ratelimit-reset header.
>
> If you exceed a secondary rate limit, you will receive a 403 or 429 response and an error message that indicates that you exceeded a secondary rate limit. If the retry-after response header is present, you should not retry your request until after that many seconds has elapsed. If the x-ratelimit-remaining header is 0, you should not retry your request until after the time, in UTC epoch seconds, specified by the x-ratelimit-reset header. Otherwise, wait for at least one minute before retrying. If your request continues to fail due to a secondary rate limit, wait for an exponentially increasing amount of time between retries, and throw an error after a specific number of retries.

Would it be possible to type `status` as `200 | ...` these other codes as well?

Also, what is the recommended pattern for handling non-200 codes with Octokit? Will `octokit.request` throw an error that we can try-catch, or do we need to check the status code manually?

(Sorry, I know that's a lot of questions)

### Versions

Octokit 22.0.1

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

Those are the GitHub API docs, but the error that `octokit.request` throws links to them when you get rate limited. For example, this is what I see in my server logs if I exceed the secondary rate limit:

> ✘ [ERROR] Uncaught (in response) Error: HttpError: You have exceeded a secondary rate limit. Please wait a few minutes before you try again. For more on scraping GitHub and how it may affect your rights, please review our Terms of Service (https://docs.github.com/en/site-policy/github-terms/github-terms-of-service) If you reach out to GitHub Support for help, please include the request ID (redacted). - https://docs.github.com/free-pro-team@latest/rest/overview/rate-limits-for-the-rest-api#about-secondary-rate-limits

That combined with the lack of guidance in the Octokit docs on how to correctly handle primary and secondary rate limit errors is why I was confused and thought I needed to check the response status code. For example, the docs say to check `response.headers['x-ratelimit-remaining']`, but that doesn't help because as you noted the request will throw a `RequestError`.

I guess I'll just follow the pattern in the `octokit/request-error` code sample.

--- a/README.md
+++ b/README.md
@@ -404,7 +404,7 @@ All other options except `options.request.*` will be passed depending on the `me
   </tr>
 </table>
 
-If an error occurs, the promise is rejected with an `error` object containing 3 keys to help with debugging:
+If an error occurs, which includes an unsuccessful status code, the promise is rejected with an `error` object containing 3 keys to help with debugging:
 
 - `error.status` The http response status code
 - `error.request` The request options such as `method`, `url` and `data`

